### DISCUSSION TITLE
Migrate from version 8 to 9

### DISCUSSION THREAD

**Discussion by yii-joblo** [directus/directus#3235] [Archive]
My current project is directus 8, vue as a frontend. 
If I want to migrate to version 9:
- will there be a document: who to to migrate?
- the database is not compatible: will there be a migration tool or some help about the difference?
- I use the sdk-js:

  `import DirectusSDK from "@directus/sdk-js";
  const client = new DirectusSDK({....});
   let articles = client.getItems('article',
        {
            fields: ['*', 'files.file.data', 'translation.*'],
            filter: {status: 'published', 'translation.article.menu.link': link}
        });`

I think the is no sdk-js in version 9. 
Who can I easily port this code to Restful or GraphQL?

It's because I like directus :-)

**Comment by rijkvanzanten** [comment]
Hi there!

> will there be a document: who to to migrate?

Yes! We're working on a couple articles right now on the differences and how to migrate.

> the database is not compatible: will there be a migration tool or some help about the difference?

The user-data (the content that you created) will be migrate-able. That being said, there are some breaking changes in the core structure of the directus tables. We've totally redesigned the permissions system, so that might be a bit hard to auto-migrate

> I think the is no sdk-js in version 9.

Yup there is! It's already live on npm too: https://www.npmjs.com/package/@directus/sdk-js 👍 

> Who can I easily port this code to Restful or GraphQL?

The SDK is nothing but a small wrapper around https://www.npmjs.com/axios, so you can fairly easily replace things like `directus.getItems('articles')` with `axios.get('/items/articles)` if you wanted to get rid of the SDK

**Comment by yii-joblo** [comment]
Thank you, I will start to migrate :-)
It's my first project using directus. The website is in progress, not online.

some other feature of my wishlist:
- tabs: [https://github.com/directus/directus/issues/1812
](https://github.com/directus/directus/issues/1812) 

- copy a whole collection

regards, joe
